### PR TITLE
Feature/sep10

### PR DESCRIPTION
**Pull request by GonzaloAlt**
# Summary

Implements SEP10 validation and sign anchor's challenge.  

* Add a node to validate the correct origin of the challenge sent by the anchor.  
* Add a node to sign the challenge from the anchor flow taking existing credentials.  

# Evidence  

</img>  
</img>  
</img>  

# References  

📖[SEP-10 documentation](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md#verification)

### CODE REVIEW

File: nodes/Anclap/transport/helpers.ts [hunk 1]
```diff
@@ -52,3 +63,84 @@ export async function buildWithdrawTransaction(
...
+		);
+
+	if (
+		webAuthDomain.length >= 1 &&
+		!webAuthDomain.every(
+			(operation) => Buffer.from(operation.value ?? '').toString('utf8') === anchorHomeDomain,
+		)
+	)
+		throw new GetChallengeValidationError(
+			'Value of operation with name "web_auth_domain" must be the Server domain',
```
Comment (Genebson) [inline nodes/Anclap/transport/helpers.ts:112]: The value of the operation with the name "web_auth_domain" must be the Server domain

File: nodes/Anclap/transport/helpers.ts [hunk 2]
```diff
@@ -52,3 +63,84 @@ export async function buildWithdrawTransaction(
...
+		);
+
+	if (operations.length > 2) {
```
Comment (Genebson) [inline nodes/Anclap/transport/helpers.ts:115]: What does this 2 means?

File: nodes/Anclap/transport/helpers.ts [hunk 3]
```diff
@@ -52,3 +63,84 @@ export async function buildWithdrawTransaction(
...
+	);
+	transaction.sign(Keypair.fromSecret(anclapCredentials.secretKey));
+
+	return { signedXdr: transaction.toXDR() };
+}
+
+function extractHomeDomain(url: string): string {
+	const domainRegex = /^(?:https?:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:\/\n?]+)/g;
+	const matches = domainRegex.exec(url);
+	if (matches && matches.length >= 2) {
```
Comment (Genebson) [inline nodes/Anclap/transport/helpers.ts:142]: Magic number.

File: nodes/Anclap/transport/helpers.ts (hunk 1 above)
Comment (GonzaloAlt) [reply nodes/Anclap/transport/helpers.ts]: Missing article added.

File: nodes/Anclap/transport/helpers.ts (hunk 3 above)
Comment (GonzaloAlt) [reply nodes/Anclap/transport/helpers.ts]: Delete as unnecessary conditional evaluation, the regular expression has only one capture group, if the regex match there will always be at least two elements in the matches array.

File: nodes/Anclap/transport/helpers.ts (hunk 2 above)
Comment (GonzaloAlt) [reply nodes/Anclap/transport/helpers.ts]: This refers to the minimum amount of operations in the challenge, I put it into a variable for higher expresiveness.

**Pull request merged**